File: serviceAccount/src/main/java/com/effective/android/service/account/AccountComponent.java
package com.effective.android.service.account;

import android.app.Application;

import com.effective.android.service.account.data.AccountRepository;
import com.plugin.component.IComponent;
import com.sina.weibo.sdk.WbSdk;
import com.sina.weibo.sdk.auth.AuthInfo;


/**
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */

public class AccountComponent implements IComponent {

    public static AccountRepository accountRepository;

    @Override
    public void attachComponent(Application application) {
        WbSdk.install(this, new AuthInfo(application, Constants.APP_KEY, Constants.REDIRECT_URL, Constants.SCOPE));
        accountRepository = new AccountRepository(application);
    }

    @Override
    public void detachComponent() {

    }

}

File: serviceAccount/src/main/java/com/effective/android/service/account/AccountComponent.kt
package com.effective.android.service.account

import android.app.Application
import com.plugin.component.IComponent
import com.plugin.component.anno.AutoInjectComponent


/**
 * Email yummyl.lau@gmail.com
 * Created by yummylau on 2018/01/25.
 */
@AutoInjectComponent(impl = [AccountServiceImpl::class])
class AccountComponent : IComponent {

    companion object {
        lateinit var sApplication: Application
    }

    override fun attachComponent(application: Application) {
        sApplication = application
        Sdks.init(application)
    }

    override fun detachComponent() {

    }
}
